File: goSolution/array/goodPair.go
package array

func numIdenticalPairs(nums []int) int {
	var rec map[int]int
	var res int
	for _, val := range nums {
		// if val, ok := rec[nums[pos]]; ok && val < pos {
		// 	res++
		// }
		rec[val]++
	}

	for _, occur := range rec {
		if occur > 1 {
			res += occur * (occur - 1) / 2
		}
	}
	return res
}

File: pythonSolution/timetrial/goodpair.py


"""
Goodpair only happens if nums[i] == nums[j] 

"""

class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        bucket = [0 * i for i in range(100)]
        res = 0
        for val in nums:
            bucket[val-1] += 1
            
        for counts in bucket:
            if counts >= 2:
                res += counts * (counts-1) / 2

        return int(res)




